### PR TITLE
Add fork := true and macro-annotations parameter

### PR DESCRIPTION
**Pull request by kellen**
Fixes spotify/scio#3960

I don't think we necessarily need to add an option for using BigQuery to the template (though we could) but we might need to improve the scio docs

### CODE REVIEW

File: src/main/g8/build.sbt [hunk 1]
```diff
@@ -32,7 +32,8 @@ lazy val commonSettings = Def.settings(
   scalacOptions ++= Seq("-target:jvm-1.8",
                         "-deprecation",
                         "-feature",
-                        "-unchecked"),
+                        "-unchecked",
+                        "-Ymacro-annotations"),
```
Comment (regadas) [inline src/main/g8/build.sbt:36]: This only applies to scala 2.13; since the template supports 2.12 as well we need to add the macro paradise compiler plugin

Comment (kellen) [reply src/main/g8/build.sbt]: Added. I also had to bump the beam-flink dep. Can I get a re-review @regadas ?

File: src/main/g8/build.sbt [hunk 2]
```diff
@@ -29,10 +29,17 @@ lazy val commonSettings = Def.settings(
   $if(DataflowRunner.truthy)$
   resolvers += "confluent" at "https://packages.confluent.io/maven/",
   $endif$
+  $if(FlinkRunner.truthy || SparkRunner.truthy)$
+  resolvers += Resolver.sonatypeRepo("releases"),
+  $endif$
```
Comment (regadas) [inline src/main/g8/build.sbt:34]: @kellen Thanks for updating this! Why is this required now?

Comment (regadas) [reply src/main/g8/build.sbt:34]: Yup this should not be needed for that.

```suggestion
```
